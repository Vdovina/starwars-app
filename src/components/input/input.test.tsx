import { create, act as testAct, ReactTestRenderer } from 'react-test-renderer';
import { render as testRender, fireEvent } from "@testing-library/react";
import { Input } from './';

describe('Input Tests', () => {
  it('Input render', () => {
    let component: ReactTestRenderer;
    testAct(() => {
      component = create(<Input className="custom-class" label="Label" placeholder='Placeholder' value="Test" onChange={() => {}} />);
    });
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Input firstIcon', () => {
    let component: ReactTestRenderer;
    testAct(() => {
      component = create(<Input firstIcon={<div>SomeNode</div>} value="Test" onChange={() => {}} />);
    });
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Input on change', () => {
    const onChange = jest.fn();
    const { container } = testRender(<Input value="Test" onChange={onChange} />);
    const input = container.querySelector<HTMLInputElement>(".input__component") as HTMLInputElement;
    fireEvent.change(input, { target: { value: 'NewValue'}});
    expect(onChange).toHaveBeenCalledTimes(1);
  });

  it('Input on clear', () => {
    const onChange = jest.fn();
    const { container } = testRender(<Input clearable value="Test" onChange={onChange} />);
    const clearButton = container.querySelector<HTMLDivElement>(".input__actions > div") as HTMLDivElement;
    fireEvent.click(clearButton);
    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
