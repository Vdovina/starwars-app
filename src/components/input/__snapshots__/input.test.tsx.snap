// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Tests Input firstIcon 1`] = `
<div
  className="input__wrapper"
>
  <div
    className="input__container"
  >
    <div
      className="input__first-icon"
    >
      <div>
        SomeNode
      </div>
    </div>
    <input
      className="input__component input__with-icon"
      onChange={[Function]}
      placeholder=""
      value="Test"
    />
    <div
      className="input__actions"
    />
  </div>
</div>
`;

exports[`Input Tests Input render 1`] = `
<div
  className="input__wrapper custom-class"
>
  <span
    className="input__label"
  >
    Label
  </span>
  <div
    className="input__container"
  >
    <input
      className="input__component"
      onChange={[Function]}
      placeholder="Placeholder"
      value="Test"
    />
    <div
      className="input__actions"
    />
  </div>
</div>
`;
